### PR TITLE
littlegptracker: 0-unstable-2020-11-26 -> 1.4.2

### LINKED ISSUE #348275: Update request: littlegptracker 0-unstable-2020-11-26 → 1.4.2
**Littlegptracker has been forked by [djdiskmachine](https://github.com/djdiskmachine/littlegptracker).**
It's being actively maintained and promises not to introduce breaking changes.
I don't insist on this fork, but I'm informing you about it :)

- Package name: `littlegptracker`
- Latest released version: _1.4.2_
- Current version on the unstable channel: `0-unstable-2020-11-26`
- Current version on the stable/release channel: `0-unstable-2020-11-26`
- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**

@fgaz

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by mrtnvgr**
This PR updates the littlegptracker package to [djdiskmachine's fork](https://github.com/djdiskmachine/littlegptracker).

Closes #348275

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by mrtnvgr** [comment]
Need testing for darwin

File: pkgs/applications/audio/littlegptracker/default.nix [hunk 1]
```diff
@@ -1,75 +1,63 @@
...
 , jack2
 , Foundation
+, alsa-lib
+, pkg-config
 }:
-
-stdenv.mkDerivation {
+let
+  python = python3.withPackages (ps: with ps; [ pillow ]);
+  platform = if stdenv.isLinux then "X64" else "OSX";
```
Comment (FliegendeWurst) [inline pkgs/applications/audio/littlegptracker/default.nix:12]:
```suggestion
  platform = if stdenv.hostPlatform.isLinux then "X64" else "OSX";
```

File: pkgs/applications/audio/littlegptracker/default.nix [hunk 2]
```diff
@@ -1,75 +1,63 @@
...
-  ++ lib.optional stdenv.hostPlatform.isDarwin Foundation
-  ++ lib.optional stdenv.hostPlatform.isLinux jack2;
-
-  patches = [
-    # Remove outdated (pre-64bit) checks that would fail on modern platforms
-    # (see description in patch file)
-    ./0001-Remove-coherency-checks.patch
-  ];
+  ++ lib.optional stdenv.isDarwin Foundation
+  ++ lib.optional stdenv.isLinux jack2;
```
Comment (FliegendeWurst) [inline pkgs/applications/audio/littlegptracker/default.nix:31]:
```suggestion
  ++ lib.optional stdenv.hostPlatform.isDarwin Foundation
  ++ lib.optional stdenv.hostPlatform.isLinux jack2;
```
Same below

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

File: pkgs/applications/audio/littlegptracker/default.nix [hunk 3]
```diff
@@ -1,75 +1,63 @@
...
 , Foundation
+, alsa-lib
+, pkg-config
 }:
-
-stdenv.mkDerivation {
+let
+  python = python3.withPackages (ps: with ps; [ pillow ]);
+  platform = if stdenv.hostPlatform.isLinux then "X64" else "OSX";
+in stdenv.mkDerivation rec {
```
Comment (fgaz) [inline pkgs/applications/audio/littlegptracker/default.nix:13]:
```suggestion
in stdenv.mkDerivation (finalAttrs: {
```

File: pkgs/applications/audio/littlegptracker/default.nix [hunk 4]
```diff
@@ -1,75 +1,63 @@
...
-  version = "0-unstable-2020-11-26";
+  version = "1.4.2";
 
   src = fetchFromGitHub {
-    owner = "Mdashdotdashn";
+    owner = "djdiskmachine";
     repo = "littlegptracker";
-    rev = "4aca8cd765e1ad586da62decd019e66cb64b45b8";
-    sha256 = "0f2ip8z5wxk8fvlw47mczsbcrzh4nh1hgw1fwf5gjrqnzm8v111x";
+    rev = "refs/tags/${version}";
```
Comment (fgaz) [inline pkgs/applications/audio/littlegptracker/default.nix:20]:
```suggestion
    rev = "refs/tags/${finalAttrs.version}";
```

File: pkgs/applications/audio/littlegptracker/default.nix [hunk 5]
```diff
@@ -1,75 +1,63 @@
...
+    hash = "sha256-1uXC5nJ63YguQuNIkuK0yx9lmrMBqw0WdlmCV8o11cE=";
   };
 
   buildInputs = [
-    SDL
+    pkg-config
+    SDL2
+    python
```
Comment (fgaz) [inline pkgs/applications/audio/littlegptracker/default.nix:27]: pkg-config and python are build-time dependencies and should go in `nativeBuildInputs`

File: pkgs/applications/audio/littlegptracker/default.nix [hunk 6]
```diff
@@ -1,75 +1,63 @@
...
   ++ lib.optional stdenv.hostPlatform.isDarwin Foundation
   ++ lib.optional stdenv.hostPlatform.isLinux jack2;
 
-  patches = [
-    # Remove outdated (pre-64bit) checks that would fail on modern platforms
-    # (see description in patch file)
-    ./0001-Remove-coherency-checks.patch
-  ];
-
   preBuild = "cd projects";
```
Comment (fgaz) [inline pkgs/applications/audio/littlegptracker/default.nix:33]: Let's format this hook properly while we're at it

```suggestion
  preBuild = ''
    cd projects
  '';
```

File: pkgs/applications/audio/littlegptracker/default.nix [hunk 7]
```diff
@@ -1,75 +1,63 @@
...
 
-  makeFlags = [ "CXX=${stdenv.cc.targetPrefix}c++" ]
-    ++ lib.optionals stdenv.hostPlatform.isLinux  [ "PLATFORM=DEB" ]
-    ++ lib.optionals stdenv.hostPlatform.isDarwin [ "PLATFORM=OSX" ];
+  makeFlags = [ "CXX=${stdenv.cc.targetPrefix}c++" "PLATFORM=${platform}" ];
 
-  env.NIX_CFLAGS_COMPILE = toString ([ "-fpermissive" ] ++
-    lib.optional stdenv.hostPlatform.isAarch64 "-Wno-error=narrowing");
+  NIX_LDFLAGS = lib.optional stdenv.isDarwin "-framework Foundation";
```
Comment (fgaz) [inline pkgs/applications/audio/littlegptracker/default.nix:37]:
```suggestion
  env.NIX_LDFLAGS = lib.optionalString stdenv.hostPlatform.isDarwin "-framework Foundation";
```

File: pkgs/applications/audio/littlegptracker/default.nix [hunk 8]
```diff
@@ -1,75 +1,63 @@
...
 
-  NIX_LDFLAGS = lib.optional stdenv.hostPlatform.isDarwin "-framework Foundation";
+  installPhase = let extension = if stdenv.isDarwin then "app" else "x64";
+    in ''
+    runHook preInstall
+    install -Dsm555 lgpt.${extension} $out/lib/lgpt/lgpt
```
Comment (fgaz) [inline pkgs/applications/audio/littlegptracker/default.nix:42]: Doesn't fixupPhase strip everything anyway (conditionally on dontStrip)?

```suggestion
    install -Dm555 lgpt.${extension} $out/lib/lgpt/lgpt
```

File: pkgs/applications/audio/littlegptracker/default.nix [hunk 9]
```diff
@@ -1,75 +1,63 @@
...
 , jack2
 , Foundation
+, alsa-lib
+, pkg-config
 }:
-
-stdenv.mkDerivation {
+let
+  python = python3.withPackages (ps: with ps; [ pillow ]);
+  platform = if stdenv.hostPlatform.isLinux then "X64" else "OSX";
```
Comment (fgaz) [inline pkgs/applications/audio/littlegptracker/default.nix:12]:
```suggestion
  platform = if stdenv.hostPlatform.isDarwin then "OSX" else "X64";
```

This way non-linux non-darwin platforms will use X64

**Review by fgaz — DISMISSED** [review]
Thanks for working on this!

Please change the first line of the commit message so it follow the conventions (`littlegptracker: 0-unstable-2020-11-26 -> 1.4.2`) and mention that we are switching to a fork in the rest of the message

File: pkgs/applications/audio/littlegptracker/default.nix [hunk 10]
```diff
@@ -1,75 +1,64 @@
...
-    sha256 = "0f2ip8z5wxk8fvlw47mczsbcrzh4nh1hgw1fwf5gjrqnzm8v111x";
+    rev = "refs/tags/${finalAttrs.version}";
+    hash = "sha256-1uXC5nJ63YguQuNIkuK0yx9lmrMBqw0WdlmCV8o11cE=";
   };
 
+  nativeBuildInputs = [ pkg-config python ];
   buildInputs = [
-    SDL
+    SDL2
+    alsa-lib
```
Comment (fgaz) [inline pkgs/applications/audio/littlegptracker/default.nix:27]:
```suggestion
  ] ++
  lib.optional (lib.meta.availableOn stdenv.hostPlatform alsa-lib) alsa-lib
```

File: pkgs/applications/audio/littlegptracker/default.nix [hunk 11]
```diff
@@ -1,75 +1,75 @@
...
-  installPhase = let extension = if stdenv.hostPlatform.isDarwin then "app" else "deb-exe";
-    in "install -Dm555 lgpt.${extension} $out/bin/lgpt";
+  env.NIX_LDFLAGS = lib.optionalString stdenv.hostPlatform.isDarwin "-framework Foundation";
 
-  passthru.updateScript = unstableGitUpdater {
-    url = "https://github.com/Mdashdotdashn/littlegptracker.git";
-  };
+  installPhase =
+    let
+      extension = if stdenv.isDarwin then "app" else "x64";
```
Comment (FliegendeWurst) [inline pkgs/applications/audio/littlegptracker/default.nix:50]:
```suggestion
      extension = if stdenv.hostPlatform.isDarwin then "app" else "x64";
```

File: pkgs/applications/audio/littlegptracker/default.nix [hunk 12]
```diff
@@ -1,75 +1,75 @@
...
+      install -Dm444 resources/${platform}/{config,mapping}.xml $out/lib/lgpt/
+      mkdir -p $out/bin
+      ln -s $out/lib/lgpt/lgpt $out/bin/
+      runHook postInstall
+    '';
 
-  meta = with lib; {
-    description = "Music tracker similar to lsdj optimised to run on portable game consoles";
+  meta = {
+    description = "Little Piggy Tracker, LittleGPTracker feature improved";
```
Comment (FliegendeWurst) [inline pkgs/applications/audio/littlegptracker/default.nix:62]: Description should not repeat the package name.
```suggestion
    description = "Music tracker optimised to run on portable game consoles";
```

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

**Comment by fgaz** [comment]
Darwin evals now, but does not build:

```
c++ -MMD -MF /private/tmp/nix-build-littlegptracker-1.4.2.drv-0/source/projects/buildOSX/Logger.d -O3 -Wall -DPLATFORM_OSX -I/private/tmp/nix-build-littlegptracker-1.4.2.drv-0/source/projects -I/Library/Frameworks/SDL.framework/Headers  -c /private/tmp/nix-build-littlegptracker-1.4.2.drv-0/source/projects/../sources/System/Console/Logger.cpp -o Logger.o
In file included from /private/tmp/nix-build-littlegptracker-1.4.2.drv-0/source/projects/../sources/System/Console/Logger.cpp:1:
In file included from /private/tmp/nix-build-littlegptracker-1.4.2.drv-0/source/projects/../sources/System/Console/Logger.h:3:
/private/tmp/nix-build-littlegptracker-1.4.2.drv-0/source/projects/../sources/System/Console/Trace.h:5:10: fatal error: 'Foundation/T_Singleton.h' file not found
#include "Foundation/T_Singleton.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~
/private/tmp/nix-build-littlegptracker-1.4.2.drv-0/source/projects/../sources/System/Console/Trace.h:5:10: note: did not find header 'T_Singleton.h' in framework 'Foundation' (loaded from '/nix/store/9l0nxr441n8qr0dk5n48m38dz4dswmg5-apple-sdk-10.12.2/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks')
```

Please either fix it or mark darwin as broken

**Review by FliegendeWurst — APPROVED** [review]

**Comment by fgaz** [comment]
Thanks!

**Pull request merged**